src/nfts/NftMedia.tsx

```diff
@@ -58,64 +67,98 @@ export default function NftImage({
   origin,
   ErrorComponent,
   testID,
+  mediaType,
 }: Props) {
   const [status, setStatus] = useState<Status>(!nft.metadata ? 'error' : 'loading')
-  const [scaledHeight, setScaledHeight] = useState(DEFAULT_IMAGE_HEIGHT)
+  const [scaledHeight, setScaledHeight] = useState(DEFAULT_HEIGHT)
   const [reloadAttempt, setReloadAttempt] = useState(0)
+  const headerHeight = useHeaderHeight()
 
   const fetchingNfts = useSelector(nftsLoadingSelector)
 
   const imageUrl = nft.media.find((media) => media.raw === nft.metadata?.image)?.gateway
+  const videoUrl = nft.media.find((media) => media.raw === nft.metadata?.animation_url)?.gateway
 
   useEffect(() => {
     if (nft.metadata) {
       setStatus('loading')
     } else {
-      sendImageLoadEvent('No nft metadata')
+      sendLoadEvent('No nft metadata')
       setStatus('error')
     }
   }, [`${nft.contractAddress}-${nft.tokenId}`])
 
+  // if the image failed to load before, try again when the user pulls to refresh
   useEffect(() => {
-    // if the image failed to load before, try again when the user pulls to refresh
     if (status === 'error' && fetchingNfts) {
       setStatus('loading')
       setReloadAttempt((prev) => prev + 1)
     }
   }, [status, fetchingNfts])
 
-  function sendImageLoadEvent(error?: string) {
+  function sendLoadEvent(error?: string) {
     const { contractAddress, tokenId } = nft
-    ValoraAnalytics.track(NftEvents.nft_image_load, {
+    ValoraAnalytics.track(NftEvents.nft_media_load, {
       tokenId,
       contractAddress,
       url: imageUrl,
       origin,
       error,
+      mediaType,
     })
 
     if (error) {
       Logger.error(
         origin,
-        `ContractAddress=${contractAddress}, TokenId: ${tokenId}, Failed to load image from ${imageUrl}`
+        `ContractAddress=${contractAddress}, TokenId: ${tokenId}, Failed to load media from ${
+          mediaType === 'video' && videoUrl ? videoUrl : imageUrl
+        }`
       )
     }
   }
 
   function handleLoadError() {
-    sendImageLoadEvent('Failed to load image')
+    sendLoadEvent('Failed to load media')
     setStatus('error')
   }
 
   function handleLoadSuccess() {
-    sendImageLoadEvent()
+    sendLoadEvent()
     setStatus('success')
   }
 
   return (
     <>
       {status === 'error' ? (
         ErrorComponent
+      ) : mediaType === 'video' && videoUrl ? (
+        <>
+          <Video
+            source={{
+              uri: videoUrl,
+              headers: {
+                origin: networkConfig.nftsValoraAppUrl,
+              },
+            }}
+            key={`${nft.contractAddress}-${nft.tokenId}-${reloadAttempt}`}
+            testID={testID}
+            style={{
+              height: DEFAULT_HEIGHT,
```

Comment:
No way to get the actual height of the video? Like we're doing with images?

Reply:
Fixed in 702a2aab1c054fcb4965d9381282a8dd77171926!